src/secureCreatives.js

```diff
@@ -94,23 +95,25 @@ function resizeRemoteCreative({ adUnitCode, width, height }) {
   });
 
   function getElementByAdUnit(elmType) {
-    let id = getElementIdBasedOnAdServer(adUnitCode);
+    let id = getElementIdBasedOnAdServer(adId, adUnitCode);
     let parentDivEle = document.getElementById(id);
     return parentDivEle && parentDivEle.querySelector(elmType);
   }
 
-  function getElementIdBasedOnAdServer(adUnitCode) {
+  function getElementIdBasedOnAdServer(adId, adUnitCode) {
     if (window.googletag) {
-      return getDfpElementId(adUnitCode)
+      return getDfpElementId(adId)
     } else if (window.apntag) {
       return getAstElementId(adUnitCode)
     } else {
       return adUnitCode;
     }
   }
 
-  function getDfpElementId(adUnitCode) {
-    return find(window.googletag.pubads().getSlots().filter(isSlotMatchingAdUnitCode(adUnitCode)), slot => slot).getSlotElementId()
+  function getDfpElementId(adId) {
+    return find(window.googletag.pubads().getSlots().filter(s =>
+      includes(s.getTargeting('hb_adid'), adId)
+    ), slot => slot).getSlotElementId();
```

Comment:
Looks like this won't work for "send all bids" where the winning bid wasn't the highest bid (which happens sometimes)

Reply:
All right! I will iterate all values then!

Reply:
@andyblackwell I've now updated the logic to search through all keys so it would find deal targeting or if something was targeted without hb_adid. Thanks for the spot!